pkg/performanceprofile/controller/performanceprofile/components/manifestset/manifestset_test.go

```diff
@@ -0,0 +1,95 @@
+package manifestset
+
+import (
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
+
+	testutils "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/utils/testing"
+)
+
+var _ = Describe("LLC Enablement file", func() {
+	var profile *performancev2.PerformanceProfile
+	var opts *components.Options
+	var expectedPath string
+
+	BeforeEach(func() {
+		profile = testutils.NewPerformanceProfile("test-llc-ann")
+		opts = &components.Options{}
+		expectedPath = "/etc/kubernetes/openshift-llc-alignment"
+	})
+
+	When("the option is configured", func() {
+		It("should not be generated if missing", func() {
+			mfs, err := GetNewComponents(profile, opts)
+			Expect(err).ToNot(HaveOccurred())
+
+			ok, err := findFileInIgnition(mfs, expectedPath)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ok).To(BeFalse(), "expected path %q found in ignition", expectedPath)
+		})
+
+		It("should not be generated if missing control annotation", func() {
```

Comment:
You should also check that the prefer-align-cpus-by-uncorecache is false in the final KubeletConfig. Otherwise the kubelet dies.